Providers/Modules/Plugins/PatchManagement/plugin/filter_patch_management.rb

```diff
@@ -30,12 +30,16 @@ def shutdown
     end
 
     def filter(tag, time, record)
-      begin
-        File.delete("/tmp/az-update-security.list")
-        OMS::Log.info_once("File deleted successfully. /tmp/az-update-security.list")
-      rescue Exception => e
-        if e.errno!=2 #except no such file or directory issue. Errno: ENOENT = 2
-          raise e
+      # checking if rb_file_path exists. if yes, skip below check. else check.
+      if File.exists?("/var/opt/microsoft/omsagent/tmp/msft-rb-path") != true
```

Comment:
This logic works if tmp folder is not deleted automatically. Are we sure this file will not be deleted?